services/github/src/handlers.js

```diff
@@ -406,14 +409,18 @@ async function statusHandler(message) {
         check_run_id: checkRun.checkRunId,
       });
     } else {
+      const taskDefinition = await this.queueClient.task(taskId).catch(debug);
+      debug(`Result status. Got task build from DB and task definition for ${taskId} from Queue service`);
+
       const checkRun = await instGithub.checks.create({
         owner: organization,
         repo: repository,
-        name: `Task ${taskId}`,
+        name: `${taskDefinition.metadata.name}: task ${taskId}`,
```

Comment:
I think I'd prefer to leave the `taskId` out of the name. It makes it quite long and hard to see the how-long-it-took portion of the reports. It is really an internal implentation detail of how taskcluster works. The user shouldn't really care or know that it exists.

I think to ensure uniqueness of `metadata.name`, we can assert that they are unique at task submission time.

Reply:
Would this work with the tasks created by a decision task as well? `jobHandler` doesn't touch those

Or are you proposing having a state (like an Azure table with all the names for particular sha or whatever)?

Reply:
Ah, you're very right about that. I definitely don't want more state. I'm reading the docs and their developer forums but don't see anything about what happens if you try to create two runs with the same title. Does it just reject the second one?

Reply:
I know what happens: the first run will get replaced by the second

Reply:
Yeah, I can't find any other solution that wouldn't include asking some DB for existing checks. Besides, it seems we are kinda pushed into that path by github... If github does it, why can't we... So be it

I'll include this into our documentation tho